docs/operator-manual/declarative-setup.md

```diff
@@ -586,6 +586,33 @@ stringData:
     }
 ```
 
+Cluster secret example using argocd-k8s-auth and [Projected Volume with a `serviceAccountToken`](https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken):
+
```

Comment:
_Not a reviewer, just some problems figuring that out when setting up OIDC-federation on my own:_ 

Maybe we can add here a sentence which service-accounts/deployments need access to the token.

The GKE example mentions: 

> Note that you must enable Workload Identity on your GKE cluster, create GCP service account with appropriate IAM role and bind it to Kubernetes service account for argocd-application-controller and argocd-server (showing Pod logs on UI). See [Use Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) and [Authenticating to the Kubernetes API server](https://cloud.google.com/kubernetes-engine/docs/how-to/api-server-authentication).

We could explicatly add something like: 
```suggestion
Cluster secret example using argocd-k8s-auth and [Projected Volume with a `serviceAccountToken`](https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken): 
Note: The projected serviceAccountToken must be available in the `argocd-application-controller` and `argocd-server` (showing Pod logs on UI) workload and the associated service account must have permissions to perform actions inside the cluster.

```